tests/tree_xml_parser_reg_text_tests.py

```diff
@@ -205,6 +205,135 @@ def test_build_from_section_table(self):
+          <SECTION>
+            <SECTNO>§ 555.219</SECTNO>
+            <SUBJECT>%s</SUBJECT>
+            <EXTRACT>
+                <HD SOURCE="HD1">%s</HD>
+                <GPOTABLE CDEF="6.1,6.1,5.2,5.2,5.2" COLS="5" OPTS="L2">
+                  <BOXHD>
+                    <CHED H="1">Pounds</CHED>
+                    <CHED H="2">Over</CHED>
+                    <CHED H="2">Not over</CHED>
+                    <CHED H="1">From inhabited building distance (feet)</CHED>
+                    <CHED H="1">From public railroad and highway distance
+                        (feet)</CHED>
+                    <CHED H="1">From above ground magazine (feet)</CHED>
+                  </BOXHD>
+                  <ROW>
+                    <ENT I="01">0</ENT>
+                    <ENT>1,000</ENT>
+                    <ENT>75</ENT>
+                    <ENT>75</ENT>
+                    <ENT>50</ENT>
+                  </ROW>
+                  <ROW>
+                    <ENT I="01">1,000</ENT>
+                    <ENT>5,000</ENT>
+                    <ENT>115</ENT>
+                    <ENT>115</ENT>
+                    <ENT>75</ENT>
+                  </ROW>
+                </GPOTABLE>
+                <HD SOURCE="HD1">%s</HD>
+            </EXTRACT>
+          </SECTION>
+        """ % (subject, table_first_header_text, table_second_header_text))
+        nodes = reg_text.build_from_section('555', xml)
+        node = nodes[0]
+        self.assertEqual(u'§ 555.219 %s' % subject, node.title)
+        self.assertEqual('regtext', node.node_type)
+        self.assertEqual(['555', '219'], node.label)
+        self.assertEqual(1, len(node.children))
+
+        extract_node = node.children[0]
+        self.assertEqual('extract', extract_node.node_type)
+        self.assertEqual(['555', '219', 'p1'], extract_node.label)
+        self.assertEqual(3, len(extract_node.children))
+
+        first_header_node = extract_node.children[0]
+        self.assertEqual('regtext', first_header_node.node_type)
+        self.assertEqual('', first_header_node.text)
+        self.assertEqual(table_first_header_text, first_header_node.title)
+        self.assertEqual(['555', '219', 'p1', 'p1'], first_header_node.label)
+        self.assertEqual(0, len(first_header_node.children))
+
+        table_node = extract_node.children[1]
+        self.assertEqual('regtext', table_node.node_type)
+        self.assertEqual('GPOTABLE', table_node.source_xml.tag)
+        self.assertEqual(['555', '219', 'p1', 'p2'], table_node.label)
+        self.assertEqual(0, len(table_node.children))
+
+        second_header_node = extract_node.children[2]
+        self.assertEqual('regtext', second_header_node.node_type)
+        self.assertEqual('', second_header_node.text)
+        self.assertEqual(table_second_header_text, second_header_node.title)
+        self.assertEqual(['555', '219', 'p1', 'p3'], second_header_node.label)
+        self.assertEqual(0, len(second_header_node.children))
```

Comment:
I'm always of two minds when it comes to this kind of thing in tests: I prefer concise code, but I also prefer tests that are really easy to read and don't require tracking through a couple of layers of abstraction to see exactly what's being tested.